src/OpenDebugAD7/Tracepoint.cs

```diff
@@ -0,0 +1,509 @@
+                            if (hr >= 0)
+                            {
+                                return pInfo[0].bstrFunction;
+                            }
+                        }
+                        return string.Format(CultureInfo.InvariantCulture, "<No Function Found>");
+                    }
+                case "$ADDRESS":
+                    {
+                        int hr = topFrame.GetCodeContext(out IDebugCodeContext2 pCodeContext);
+                        if (hr >= 0)
+                        {
+                            CONTEXT_INFO[] pInfo = new CONTEXT_INFO[1];
+                            hr = pCodeContext.GetInfo(enum_CONTEXT_INFO_FIELDS.CIF_ADDRESS, pInfo);
+                            if (hr >= 0)
+                            {
+                                return pInfo[0].bstrAddress;
+                            }
+                        }
+                        return string.Format(CultureInfo.InvariantCulture, "<No Address Found>");
+                    }
+                case "$FILEPOS":
+                    {
+                        int hr = topFrame.GetDocumentContext(out IDebugDocumentContext2 pDocumentContext);
+                        if (hr >= 0)
+                        {
+                            hr = pDocumentContext.GetName(enum_GETNAME_TYPE.GN_FILENAME, out string fileName);
+                            if (hr >= 0)
+                            {
+                                TEXT_POSITION[] textPosBeg = new TEXT_POSITION[1];
+                                TEXT_POSITION[] textPosEnd = new TEXT_POSITION[1];
+                                hr = pDocumentContext.GetStatementRange(textPosBeg, textPosEnd);
+                                if (hr >= 0)
+                                {
+                                    return string.Format(CultureInfo.InvariantCulture, "{0}({1})", fileName, textPosBeg[0].dwLine + 1);
+                                }
+                            }
+                        }
+                        return string.Format(CultureInfo.InvariantCulture, "<No File Posiition>");
+                    }
+                case "$CALLER":
+                    {
+                        IEnumDebugFrameInfo2 frameInfoEnum;
+                        int hr = pThread.EnumFrameInfo(enum_FRAMEINFO_FLAGS.FIF_FRAME | enum_FRAMEINFO_FLAGS.FIF_FLAGS, Constants.EvaluationRadix, out frameInfoEnum);
+                        if (hr >= 0)
+                        {
+                            FRAMEINFO[] frames = new FRAMEINFO[2];
+                            uint fetched = 0;
+                            hr = frameInfoEnum.Next(2, frames, ref fetched);
+                            if (hr < 0 && fetched == 2)
+                            {
+                                return frames[1].m_bstrFuncName;
+                            }
+                        }
+                        return string.Format(CultureInfo.InvariantCulture, "<No caller avaliable>");
```

Comment:
why do your other errors have the first letter of each word capitalized but not here?